.github/workflows/docker.yaml

```diff
@@ -0,0 +1,77 @@
+name: Docker Image CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*"
+
+jobs:
+  build:
+    name: Build and publish docker images
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        arrays: [
+          {name: 'loadtester', file:'Dockerfile.load-tester'},
+          {name: 'record-tester', file:'Dockerfile.record-tester'},
+          {name: 'orch-tester', file:'Dockerfile.orch-tester'},
+          {name: 'mist-api-connector', file:'Dockerfile.mist api-connector'},
+          {name: 'streamtester', file:'Dockerfile'},
+        ]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          # Check https://github.com/livepeer/go-livepeer/pull/1891
+          # for ref value discussion
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: livepeer/${{ matrix.arrays.name }}
+          tags: |
+            type=schedule,pattern={{date 'YYYYMMDDHHmmss'}}
+            type=ref,event=branch
+            type=ref,event=pr
+            type=sha
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+
+      - name: Tags
+        id: tags
+        uses: livepeer/action-gh-release-tags@v0
+        with:
+          always-latest-on-branch: master
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          build-args: |
+            version=${{ steps.tags.outputs.version }}
```

Comment:
Ooh, so we need the tags step only for the version build arg now?

Reply:
yeah! i couldn't find an alternative to this yet :(

Reply:
yeah! i couldn't find an alternative to this yet :(